src/MLOps.NET/Catalogs/ModelCatalog.cs

```diff
@@ -0,0 +1,29 @@
+﻿using MLOps.NET.Storage;
+using System;
+using System.Threading.Tasks;
+
+namespace MLOps.NET.Catalogs
+{
+    /// <summary>
+    /// Operations related to a model
+    /// </summary>
+    public sealed class ModelCatalog
+    {
+        private readonly IModelRepository modelRepository;
+
+        /// <summary>
+        /// ctor
+        /// </summary>
+        /// <param name="modelRepository"></param>
+        public ModelCatalog(IModelRepository modelRepository)
+        {
+            this.modelRepository = modelRepository;
+        }
+
+        ///<inheritdoc/>
+        public async Task UploadModelAsync(Guid runId, string filePath)
```

Comment:
Just `UploadAsync` will do? `mlContext.Model.UploadAsync`

Reply:
I like the suggestion, makes a lot of sense now when it's on the specific `ModelCatalog`. Pushed a change for it.